### ISSUE TITLE
[no-restricted-exports] Fails on `export { default } from ...`

### ISSUE THREAD

**Issue by milesj** [airbnb/javascript#2500]
As the title states. Fails when re-exporting the default from another module.

This feels like it _should_ pass, since this is not a named export. https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/es6.js#L67

**Comment by ljharb** [comment]
You're right, it definitely should be acceptable. This seems like a bug in eslint itself, given https://eslint.org/docs/rules/no-restricted-exports#default-exports

Can you file that on eslint?

**Comment by alekslario** [comment]
bump

**Comment by tincho** [comment]
just FTR, a quick workaround for me was adding

```json
        "no-restricted-exports": ["error", {
          "restrictedNamedExports": [
            "then"
          ]
        }],
```

to `rules` object in my `.eslintrc.json` file

it would have been quicker to just add 

```json
        "no-restricted-exports": "off"
```

but I just wanted to remove the `default` keyword and keep the rest of the restricted

**Comment by elyobo** [comment]
https://github.com/eslint/eslint/issues/15617 has now resolved https://github.com/eslint/eslint/issues/15617 with a fix available upstream by disabling `defaultFrom` for `no-restricted-exports`. @ljharb would you accept a PR changing this config in the rules in this repo?

**Comment by ljharb** [comment]
@elyobo if we enabled the option, it would break on eslint versions older than the one that added it - so, it's a breaking change to do so. This means that there should be a "TODO" comment above the rule, describing it as semver-major, and what change to make - that way, when we next _have_ a breaking change, I'll do a sweep through the comments for any rules that are waiting to be enabled.

**Comment by elyobo** [comment]
OK, thanks for the explanation, I've opened https://github.com/airbnb/javascript/pull/2721.

I've also flagged on https://github.com/eslint/eslint/pull/16785#issuecomment-1413236227 but it looks like the change included arguably breaking changes to the option validating by banning the string `default` in `restrictedNamedExports` as is used in the airbnb configs. This might bite users of existing airbnb config releases that ugprade to eslint 8.33.0+

**Comment by ljharb** [comment]
Thanks, that's great.
